### ISSUE TITLE
Cannot convert a datetimelike to 'str' with astype

### ISSUE THREAD

**Issue by wavexx** [pydata/pandas#9757]
I find this surprising:

    import pandas as pd
    pd.Series([pd.Timestamp('2010-01-04 00:00:00')]).astype(str)
    TypeError: cannot astype a datetimelike from [datetime64[ns]] to [|S0]

str(pd.Timestamp) works though, so why astype shouldn't?
Also the type representation "|S0" is not entirely clear to me.

**Comment by jreback** [comment]
this is string formatting and not what astype does which is convert to a fixed width string dtype which is not the same as a str representation

that's why this is not allowed 
further there are many possible formats for strings
use .format()

it's possible that astype should just do the default formatting but I'd like to see some justification for this which addresses the points above

**Comment by evanpw** [comment]
This conversion is explicitly allowed for every other type (e.g., conversion to string is checked for directly in ``core.common._astype_nansafe``).

In numpy, a call to ``astype(str)`` converts to a fixed-width string with a size large enough to contain every entry:

```
In [2]: np.array([np.datetime64('2010-01-03 00:00:00')]).astype(str)
Out[2]:
array(['2010-01-03T05:00:00Z'],
      dtype='|S38')
```

That might be a better choice, but it's not what's currently being done for types other than ``datetime64`` and ``timedelta64``:

```
In [3]: pd.Series([1.5]).astype(str)
Out[3]:
0    1.5
dtype: object
```

I actually can't get ``astype`` to convert to a fixed-width string even if I try this:
```
In [2]: pd.Series([1.5]).astype('S10')
Out[2]:
0    1.5
dtype: object
```

**Comment by wavexx** [comment]
I personally expect conversion to 'str' to always work and default to __str__.
I know format it's an option, but it shouldn't be mandatory.

I was trying to convert an entire DataFrame to 'str' (for quick displaying purposes, nothing fancy).
It's worth nothing that:

    pd.Series([pd.Timestamp('2010-01-04 00:00:00')]).astype(object).astype(str)

does what I had in mind but adds an useless conversion.

**Comment by evanpw** [comment]
Actually, it looks like I'm confused about fixed-width string dtypes. Is it even possible to store such a thing in a Series or DataFrame? I can't seem to find any way.

**Comment by jreback** [comment]
fixed width string dtypes are not supported and converted to object

**Comment by shoyer** [comment]
It's true that we don't support fixed width string dtypes, but it would be nice for `.astype(str)` to reliably coerce to an object array of strings.

**Comment by jorisvandenbossche** [comment]
I also see no reason to not let `astype(str)` convert it to an object array of strings with default formatting.

Apart from that, we could maybe also use a separate formatting function (similar to ``strftime`` for datetime.datetime)

**Comment by jreback** [comment]
closed by #9758
